lumaviewpro.py

```diff
@@ -3279,23 +3281,26 @@ def insert_step(self, after_current_step: bool = True):
         else:
             after_step = None
             before_step = self.curr_step
+        
+        for acquired_layer in acquired_layers:
+            acquired_layer_config = get_layer_configs()[acquired_layer]
+
+            step_name = self._protocol.insert_step(
+                step_name=None,
+                layer=acquired_layer,
+                layer_config=acquired_layer_config,
+                plate_position=plate_position,
+                objective_id=objective_id,
+                before_step=before_step,
+                after_step=after_step
+            )
 
-        step_name = self._protocol.insert_step(
-            step_name=None,
-            layer=active_layer,
-            layer_config=active_layer_config,
-            plate_position=plate_position,
-            objective_id=objective_id,
-            before_step=before_step,
-            after_step=after_step
-        )
-
-        if after_current_step or (self.curr_step < 0):
-            self.curr_step += 1
+            if after_current_step or (self.curr_step < 0):
+                self.curr_step += 1
 
-        self.update_step_ui()
-        stage.set_protocol_steps(df=self._protocol.steps())
-        self.go_to_step()
+            self.update_step_ui()
+            stage.set_protocol_steps(df=self._protocol.steps())
+            self.go_to_step()
```

Comment:
I think these 3 lines should have stayed the same.  They can all be executed once, after all the new steps have been added to the protocol.